demos/fleet_provisioning/fleet_provisioning_with_csr/mqtt_operations.c

```diff
@@ -1007,17 +1118,30 @@ bool PublishToTopic( const char * pTopicFilter,
 }
 /*-----------------------------------------------------------*/
 
-bool ProcessLoop( void )
+bool ProcessLoopWithTimeout( void )
 {
+    uint32_t ulMqttProcessLoopTimeoutTime;
+    uint32_t ulCurrentTime;
+
+    MQTTStatus_t eMqttStatus = MQTTSuccess;
     bool returnStatus = false;
-    MQTTStatus_t mqttStatus = MQTTSuccess;
 
-    mqttStatus = MQTT_ProcessLoop( &mqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+    ulCurrentTime = mqttContext.getTime();
+    ulMqttProcessLoopTimeoutTime = ulCurrentTime + MQTT_PROCESS_LOOP_TIMEOUT_MS;
 
-    if( mqttStatus != MQTTSuccess )
+    /* Call MQTT_ProcessLoop multiple times until the timeout expires or
+     * #MQTT_ProcessLoop fails. */
+    while( ( ulCurrentTime < ulMqttProcessLoopTimeoutTime ) &&
+           ( eMqttStatus == MQTTSuccess || eMqttStatus == MQTTNeedMoreBytes ) )
+    {
+        eMqttStatus = MQTT_ProcessLoop( &mqttContext );
+        ulCurrentTime = mqttContext.getTime();
```

Comment:
Here too. Should we avoid a busy wait loop?

Reply:
The busy wait loop is because the demo is expecting incoming publishes ... and it hinged on MQTT_ProcessLoops timeout before...